patches/tModLoader/Terraria/ModLoader/UI/UIModItem.cs

```diff
@@ -371,5 +393,82 @@ internal class UIModItem : UIPanel
 					return !_mod.Enabled;
 			}
 		}
+
+		private void QuickModDelete(UIMouseEvent evt, UIElement listeningElement) {
+			bool shiftPressed = false;
+			List<Keys> pressedKeys = Main.keyState.GetPressedKeys().ToList();
+			for (int i = 0; i < pressedKeys.Count; i++) {
+				if (pressedKeys[i] == Keys.LeftShift || pressedKeys[i] == Keys.RightShift) {
+					shiftPressed = true;
+					break;
+				}
+			}
+
+			if (!shiftPressed) {
+				SoundEngine.PlaySound(10, -1, -1, 1);
+				_blockInput = new UIImage(TextureAssets.Extra[190]) {
+					Width = { Percent = 1 },
+					Height = { Percent = 1 },
+					Color = new Color(0, 0, 0, 0),
+					ScaleToFit = true
+				};
+				_blockInput.OnMouseDown += CloseDialog;
+				Interface.modsMenu.Append(_blockInput);
+
+				_deleteModDialog = new UIPanel() {
+					Width = { Percent = .20f },
+					Height = { Percent = .20f },
+					HAlign = .5f,
+					VAlign = .5f,
+					BackgroundColor = new Color(63, 82, 151),
+					BorderColor = Color.Black
+				};
+				_deleteModDialog.SetPadding(6f);
+				Interface.modsMenu.Append(_deleteModDialog);
+
+				_dialogYesButton = new UIAutoScaleTextTextPanel<string>(Language.GetTextValue("tModLoader.Yes")) {
+					TextColor = Color.White,
+					Width = new StyleDimension(-10f, 1f / 3f),
+					Height = { Pixels = 40 },
+					VAlign = .85f,
+					HAlign = .15f
+				}.WithFadedMouseOver();
+				_dialogYesButton.OnClick += DeleteMod;
+				_deleteModDialog.Append(_dialogYesButton);
+
+				_dialogNoButton = new UIAutoScaleTextTextPanel<string>(Language.GetTextValue("tModLoader.No")) {
+					TextColor = Color.White,
+					Width = new StyleDimension(-10f, 1f / 3f),
+					Height = { Pixels = 40 },
+					VAlign = .85f,
+					HAlign = .85f
+				}.WithFadedMouseOver();
+				_dialogNoButton.OnClick += CloseDialog;
+				_deleteModDialog.Append(_dialogNoButton);
+
+				_dialogText = new UIText(Language.GetTextValue("tModLoader.DeleteModConfirm")) {
+					Width = { Percent = .75f },
+					HAlign = .5f,
+					VAlign = .3f,
+					IsWrapped = true
+				};
+				_deleteModDialog.Append(_dialogText);
+			}
+			else {
+				DeleteMod(evt, listeningElement);
+			}
+		}
+
+		private void CloseDialog(UIMouseEvent evt, UIElement listeningElement) {
+			SoundEngine.PlaySound(SoundID.MenuClose);
+			_blockInput?.Remove();
+			_deleteModDialog?.Remove();
+		}
+
+		private void DeleteMod(UIMouseEvent evt, UIElement listeningElement) {
+			File.Delete(_mod.modFile.path);
```

Comment:
You must call `_mod.Close()` before deleting the tmod file. To allow the mod to close any open file handles (like streaming music).

You must also reload immediately after exiting the mods menu.

Reply:
The trash icon only shows up on disabled mods, shouldn't all mods have already no open file handles? I can try to make it work for enabled mods too, although I don't think it makes much sense to uninstall enabled mods.

Reply:
Regarding the inventory trash icon, that's the one I'm using, except without a background border. ![img](https://user-images.githubusercontent.com/30240961/106349389-73741980-62c5-11eb-9c29-0ae2c5287752.png)

Not sure if there's a blank border that I can add below it to make it look like all the other buttons.